app/src/main/java/com/gophillygo/app/FilterDialog.java

```diff
@@ -55,15 +66,31 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         View contentView = View.inflate(getContext(), R.layout.filter_modal, null);
         dialog.setContentView(contentView);
 
+        Filter filter = null;
+        if (getArguments() != null) {
+            filter = getArguments().getParcelable(FILTER_ARG);
+        }
+        if (filter == null) {
+            filter = new Filter();
+        }
+
         // set up filter buttons
         natureButton = dialog.findViewById(R.id.filter_modal_nature_category_button);
+        natureButton.setChecked(filter.nature);
```

Comment:
Should switch to data binding for these and remove all this `findViewById` boilerplate.